airflowHPC/dags/replex.py

```diff
@@ -233,14 +230,14 @@ def propose_swap(swappables):
     return swap
 
 
-def calc_prob_acc(swap, dhdl_files, states):
+def calc_prob_acc(swap, dhdl_files, states, shifts):
```

Comment:
Hey Joe! Is there a specific reason that you prefer to make `shifts` as an input argument even if it can be calculated inside the function `shifts = [SHIFT_RANGE for i in range(NUM_SIMULATIONS)]`?

Reply:
The reason I opened up a new MR is that in testing of my existing MR I noticed that I sometimes got `IndexError` in the `calc_prob_acc` function, so I decided to start over from the MR you made (#7). I then noticed that this also suffered sometimes from `IndexError`'s. I couldn't figure out any issues on comparing with your code in ensemble_md so I used your tests of the `calc_prob_acc` function from your ensemble_md repo, which required a change of signature. The tests turned out to be very useful! In the end I found that in your tests you had `NUM_STATES` (or `self.n_tot` in ensemble_md) set to 9 whereas in #7 you had set it to 8. This bug took quite a while to track down, since I assumed that the constants were set correctly, but the important thing is that your tests help track it down.

I just have a few minor changes and I think this MR will be ready to merge.

N.B. this shows why it is probably better anyway to pass the shifts as an argument to`calc_prob_acc`: global variables make testing more challenging both because one tends to assume they are set correctly and also they make it harder to see all the inputs a function needs.

Reply:
Also `shifts = [SHIFT_RANGE for i in range(NUM_SIMULATIONS)]`, which would give `[1, 1, 1, 1]` with `SHIFT_RANGE=1` and `NUM_SIMULATIONS=4`, does not match what you have in ensemble_md. There you have `shifts = list(SHIFT_RANGE * np.arange(NUM_SIMULATIONS))`, which gives `[1, 2, 3, 4]`. I think you might have meant to type `shifts = [SHIFT_RANGE * i for i in range(NUM_SIMULATIONS)]` but left out the `* i`. For this an IDE like PyCharm or VSCode that will highlight unused variables can be helpful.

Reply:
> In the end I found that in your tests you had NUM_STATES (or self.n_tot in ensemble_md) set to 9 whereas in https://github.com/ejjordan/airflowHPC/pull/7 you had set it to 8.

Ugh I see! Sorry that I set the constant incorrectly! 

> global variables make testing more challenging both because one tends to assume they are set correctly and also they make it harder to see all the inputs a function needs.

Hmm interestingly in `ensemble_md`, `calc_prob_acc` did take in `shifts` as an input argument, but I don't quite remember why I took it down in `airflowHPC`. And you are right that it should be `list(SHIFT_RANGE * np.arange(NUM_SIMULATIONS))`, or equivalently (in terms of the output), `shifts = [SHIFT_RANGE * i for i in range(NUM_SIMULATIONS)]`. I must have been careless when transferring/translating the codes from `ensemble_md` ...

Reply:
No worries, in the end the solution was not much work. We all make mistakes. Hence the value of having tests!

As an aside I might add that if you were to design the main class in ensemble_md again, you might have a proper init function that sets the member properties and values and a factory function that calls the init function after unpacking the values from a yaml (or other) file. This would make it easier to see where the member values come from (and would mean that clicking through to where a value is set in an IDE works). In particular this would make the input values used in testing more legible.

Reply:
> you might have a proper init function that sets the member properties and values and a factory function that calls the init function after unpacking the values from a yaml (or other) file.

That's a good point. I did consider refactoring the code in `ensemble_md` at some point. Though that might be of lower priority now, I'll take this into consideration in the future when refactoring!

Reply:
Yeah just something to keep in mind when designing classes in the future.